piff/stats.py

```diff
@@ -0,0 +1,264 @@
+        # for position in positions:
+        #     print(position)
+        #     stars_model.append(self.generate_star(position))
+        stars_model = np.array([ self.generate_star(position) for position in positions ])
+        # measure moments with Gaussian on interpolated model image
+        if logger:
+            logger.info("Measuring Model Stars")
+        shapes_model = np.array([ hsm.fitStar(star).getParameters() for star in stars_model ])
+
+        # define terms for the catalogs
+        ra = positions[:, 0]
+        dec = positions[:, 1]
+        T = shapes_truth[:, 0]
+        g1 = shapes_truth[:, 1]
+        g2 = shapes_truth[:, 2]
+        dT = T - shapes_model[:, 0]
+        dg1 = g1 - shapes_model[:, 1]
+        dg2 = g2 - shapes_model[:, 2]
+
+        # make the treecorr catalogs
+        corr_dict = {'min_sep': min_sep, 'max_sep': max_sep,
+                     'bin_size': bin_size, 'sep_units': sep_units,
+                     }
+        if logger:
+            logger.info("Creating Treecorr Catalogs")
+
+        if sky_coordinates:
+            cat_g = treecorr.Catalog(ra=ra, dec=dec,
+                                     g1=g1, g2=g2,
+                                     ra_units=ra_units, dec_units=dec_units)
+            cat_dg = treecorr.Catalog(ra=ra, dec=dec,
+                                      g1=dg1, g2=dg2,
+                                      ra_units=ra_units, dec_units=dec_units)
+            cat_gdTT = treecorr.Catalog(ra=ra, dec=dec,
+                                        g1=g1 * dT / T, g2=g2 * dT / T,
+                                        ra_units=ra_units, dec_units=dec_units)
+        else:
+            cat_g = treecorr.Catalog(x=ra, y=dec,
+                                     g1=g1, g2=g2)
+            cat_dg = treecorr.Catalog(x=ra, y=dec,
+                                      g1=dg1, g2=dg2)
+            cat_gdTT = treecorr.Catalog(x=ra, y=dec,
+                                        g1=g1 * dT / T, g2=g2 * dT / T)
+
+        # setup and run the correlations
+        if logger:
+            logger.info("Processing rho PSF statistics")
+
+        # save the rho objects
+        rho1 = treecorr.GGCorrelation(**corr_dict)
+        rho1.process(cat_dg)
+        rho2 = treecorr.GGCorrelation(**corr_dict)
+        rho2.process(cat_g, cat_dg)
+        rho3 = treecorr.GGCorrelation(**corr_dict)
+        rho3.process(cat_gdTT)
+        rho4 = treecorr.GGCorrelation(**corr_dict)
+        rho4.process(cat_dg, cat_gdTT)
+        rho5 = treecorr.GGCorrelation(**corr_dict)
+        rho5.process(cat_g, cat_gdTT)
+
+        # save the rhos for later
+        self.rho = {1: rho1,
+                    2: rho2,
+                    3: rho3,
+                    4: rho4,
+                    5: rho5}
+
+        # return directly the correlation functions
+        return rho1.logr, rho1.xip, rho2.xip, rho3.xip, rho4.xip, rho5.xip
+
+    def plot(self, rho, fig=None, ax=None, logger=None, **kwargs):
```

Comment:
Useful to have something that returns plots so they can be displayed from within a python session.  However, more often, we will probably want to write this to a file.  So maybe have a `write()` method in the Statistics base class that will take the output a `stat.plot(...)` and write it to a a file.